cicecore/cicedynB/general/ice_init.F90

```diff
@@ -2657,99 +2893,255 @@ subroutine set_state_var (nx_block, ny_block, &
-                  ainit(n) = p5 !echmod symm
+                  ainit(n) = abar
                endif
             enddo
 
-         elseif (trim(ice_data_type) == 'boxslotcyl') then
-         
-            hbar = c1  ! initial ice thickness (1 m)
+         elseif (trim(ice_data_conc) == 'parabolic') then
+
+            ! initial category areas in cells with ice
+            hbar = c3  ! initial ice thickness with greatest area
+                       ! Note: the resulting average ice thickness 
+                       ! tends to be less than hbar due to the
+                       ! nonlinear distribution of ice thicknesses 
+            sum = c0
             do n = 1, ncat
-               hinit(n) = c0
-               ainit(n) = c0
-               if (hbar > hin_max(n-1) .and. hbar < hin_max(n)) then
-                  hinit(n) = hbar
-                  ainit(n) = c1 !echmod symm
+               if (n < ncat) then
+                  hinit(n) = p5*(hin_max(n-1) + hin_max(n)) ! m
+               else                ! n=ncat
+                  hinit(n) = (hin_max(n-1) + c1) ! m
                endif
+               ! parabola, max at h=hbar, zero at h=0, 2*hbar
+               ainit(n) = max(c0, (c2*hbar*hinit(n) - hinit(n)**2))
+               sum = sum + ainit(n)
             enddo
-         
+            do n = 1, ncat
+               ainit(n) = ainit(n) / (sum + puny/ncat) ! normalize
+            enddo
+
          else
 
-      ! initial category areas in cells with ice
-         hbar = c3  ! initial ice thickness with greatest area
-                    ! Note: the resulting average ice thickness 
-                    ! tends to be less than hbar due to the
-                    ! nonlinear distribution of ice thicknesses 
-         sum = c0
-         do n = 1, ncat
-            if (n < ncat) then
-               hinit(n) = p5*(hin_max(n-1) + hin_max(n)) ! m
-            else                ! n=ncat
-               hinit(n) = (hin_max(n-1) + c1) ! m
-            endif
-            ! parabola, max at h=hbar, zero at h=0, 2*hbar
-            ainit(n) = max(c0, (c2*hbar*hinit(n) - hinit(n)**2))
-            sum = sum + ainit(n)
-         enddo
-         do n = 1, ncat
-            ainit(n) = ainit(n) / (sum + puny/ncat) ! normalize
-         enddo
+            call abort_ice(subname//'ERROR: ice_data_conc setting = '//trim(ice_data_conc), &
+               file=__FILE__, line=__LINE__)
 
-         endif ! ice_data_type
+         endif ! ice_data_conc
 
-         if (trim(grid_type) == 'rectangular') then
+         !---------------------------------------------------------
+         ! location of ice
+         !---------------------------------------------------------
 
-         ! place ice on left side of domain
-         icells = 0
-         do j = jlo, jhi
-         do i = ilo, ihi
-            if (tmask(i,j)) then
-               if (ULON(i,j) < -50./rad_to_deg) then
+         if (trim(ice_data_type) == 'box2001') then
```

Comment:
Of all these comments, this is the only one that I'm concerned enough about to want more information.  In particular, I'd like to make sure that the box2001 test produces the same results (not necessarily BFB but close) as it would have in the original paper with or without the islands (depending on whether they are included in the land mask).  I'd like to understand exactly what the bug fixes were.  Can you point me to the specific lines of code?  
If the box2001 results are qualitatively the same, then I'm fine with merging this PR now and doing a bunch of cleanup later.

Reply:
The main thing that changed in box2001 was a bug fix to the ocean current forcing (old line 5291, new line 5460, note denominator).  But that should not impact square grids.    The other thing that was changed was that strax/stray have aice in the equation, not aiu (aice mapped to u points), that's line 5883 in ice_forcing.  That would change results.

I'm happy to do a clean comparison of the box2001 results on main and on cgridDEV.  Our standard test is gbox80 on 1 pe for 10 days.  Do we need a longer run?  What diagnostics do you want to see?

Reply:
I put some plots on one of the issues, https://github.com/CICE-Consortium/CICE/issues/694#issuecomment-1121691861.

Reply:
Thanks @apcraig .  Those results are sufficiently similar to satisfy my concern.